plugins/catalog-react/src/components/EntityOwnerPicker/useFacetsEntities.ts

```diff
@@ -57,33 +58,40 @@ export function useFacetsEntities({ enabled }: { enabled: boolean }) {
     });
     const entityRefs = facetsResponse.facets[facet]?.map(e => e.value) ?? [];
 
-    return catalogApi
-      .getEntitiesByRefs({ entityRefs })
-      .then(resp =>
-        resp.items
-          .filter(entity => entity !== undefined)
-          .map(entity => entity as Entity)
-          .sort(
-            (a, b) =>
-              (a.metadata.namespace || '').localeCompare(
-                b.metadata.namespace || '',
-                'en-US',
-              ) ||
-              (
-                maybeString(get(a, 'spec.profile.displayName')) ||
-                a.metadata.title ||
-                a.metadata.name
-              ).localeCompare(
-                maybeString(get(b, 'spec.profile.displayName')) ||
-                  b.metadata.title ||
-                  b.metadata.name,
-                'en-US',
-              ) ||
-              a.kind.localeCompare(b.kind, 'en-US'),
-          ),
-      )
-      .then(entities => entities)
-      .catch(() => []);
+    // Fetch entities in chunks, because there's a limit to how many refs that
+    // can be fetched at once before hitting "413 Payload Too Large" limits of
+    // the typical catalog installation
+    const entities = new Array<Entity | undefined>();
+    for (const refsChunk of chunk(entityRefs, 1000)) {
+      Array.prototype.push.apply(
+        entities,
+        await catalogApi
+          .getEntitiesByRefs({ entityRefs: refsChunk })
+          .then(r => r.items)
+          .catch(() => []),
+      );
```

Comment:
`entities.push(...newEntities)` plz

Reply:
But performance!

Hehe, maybe it turns out to be the same thing behind the scenes anyway

Reply:
Fixed